### DISCUSSION TITLE
How do I generate an example RA-TLS certificate in as few steps as possible?

### DISCUSSION THREAD

**Discussion by fwoodruff** [gramineproject/gramine#1649] [General]
Running `gramine-ratls ./cert.pem ./key.pem` in an empty directory I get:
`ra_tls_create_key_and_crt_der returned -10496`

Going to the following page: https://gramine.readthedocs.io/en/stable/manpages/gramine-ratls.html#example
I can see a manifest file. What do I do with it? Alternatively, can anyone point me to an example RA-TLS certificate? I don't even need the key. All I am trying to do is inspect the extensions

**Comment by dimakuv** [comment]
> Going to the following page: https://gramine.readthedocs.io/en/stable/manpages/gramine-ratls.html#example
I can see a manifest file. What do I do with it?

If you don't know what to do with a Gramine manifest file, you're probably very new to the project.

I recommend you to play with Gramine a bit, to understand how Gramine works and what Gramine manifest files are used for. You can start with this video: https://archive.fosdem.org/2022/schedule/event/tee_gramine/

And then proceed to e.g. a Helloworld example: https://github.com/gramineproject/gramine/tree/master/CI-Examples/helloworld

After getting some understanding on how Gramine build and runtime flows work, you can e.g. modify the `helloworld.manifest.template` file according the the `gramine-ratls` example, rebuild using Make, and see how it works.
